### PR TITLE
[Facebook Custom Audiences] Sync users to an audience

### PR DESCRIPTION
**Pull request by nick-Ag**
This PR adds the core functionality of syncing audiences to the Facebook Custom Audiences destination.

## Testing
[Stage Testing Doc](https://docs.google.com/document/d/1qY3CRucuxHzPpVOb1_LduXhYxrAabX-3rJdZ633fjPQ/edit)
Tested successfully in staging by syncing a RETL warehouse. 

Small Batch of 2 records successfully syncing:
Screenshot 2024-07-31 at 3 12 27 PM

That same batch correctly arriving in Facebook:
Screenshot 2024-07-31 at 3 12 46 PM

Larger batch of 8 records successfully syncing:
Screenshot 2024-07-31 at 3 22 07 PM

The batch arriving in Facebook:
Screenshot 2024-07-31 at 3 21 50 PM

### Removal
Successfully tested the delete flow:

2 Records removed:

Screenshot 2024-08-02 at 11 24 15 AM

Those 2 records removed in Facebook:
Screenshot 2024-08-02 at 11 24 09 AM

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/facebook-custom-audiences/__tests__/fb-operations.test.ts [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 import nock from 'nock'
+import { Payload } from '../sync/generated-types'
+import { createHash } from 'crypto'
+import { normalizationFunctions } from '../fbca-properties'
 
 const requestClient = createRequestClient()
 const settings: Settings = {
   retlAdAccountId: 'act_123456'
 }
+const EMPTY = ''
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/facebook-custom-audiences/__tests__/fb-operations.test.ts:13]: ha, nice touch

File: packages/destination-actions/src/destinations/facebook-custom-audiences/constants.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+export const US_STATE_CODES = new Map<string, string>([
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/facebook-custom-audiences/constants.ts:1]: Where did you get these from? Maybe add a link to the reference you used?

**Review by marinhero — COMMENTED** [review]
Overall, this looks pretty good. Let's move ahead to test it. Update your test doc and provide evidence so I can approve this.

**Comment by nick-Ag** [comment]
> Overall, this looks pretty good. Let's move ahead to test it. Update your test doc and provide evidence so I can approve this.

Linked Test doc in description + Added some evidence for the delete operation working

**Review by marinhero — APPROVED** [review]

**Pull request merged**